custom_scripts/head.liquid

```diff
@@ -1,58 +1,216 @@
+      <script>
+              window.BM.on('ready', () => {
+        window.BM.scriptManager.register({
+          name: 'google-analytics-legacy',
+          consentCategory: 'analytics',
+          load: () => {
+            // Load script
+            (function(i, s, o, g, r, a, m) {
+              i['GoogleAnalyticsObject'] = r;
+              i[r] = i[r] || function() {
+                  (i[r].q = i[r].q || []).push(arguments)
+              }, i[r].l = 1 * new Date();
+              a = s.createElement(o),
+                  m = s.getElementsByTagName(o)[0];
+              a.async = 1;
+              a.src = g;
+              m.parentNode.insertBefore(a, m)
+            })(window, document, 'script', 'https://www.google-analytics.com/analytics.js', 'ga');
+
+            // Initialize Google Analytics
+            ga('create', '{{ settings.measurementId }}', 'auto');
+            {% if anonymize -%}
+              ga('set', 'anonymizeIp', true);
+            {% endif -%}
+            
+            ga('set', 'userId', '{{ member.id }}');
+
+            // Listen for page views
+            const onPageView = (data) => {
+              ga('send', {
+                'hitType': 'pageview',
+                'page': data.page,
+                'title': data.title,
+              });
+            }
+            if (window.Tribe) {
+              window.ga4PageViewListener = window.Tribe.on('pageView', onPageView);
+            }
+          },
+          unload: () => {
+            // Remove script
+            const script = document.querySelector('script[src*="google-analytics.com/analytics.js"]');
+            if (script) {
+              script.remove();
+            }
+
+            // Remove dataLayer
+            window.dataLayer = undefined;
+
+            // Remove page view listener
+            if (window.ga4PageViewListener) {
+              window.ga4PageViewListener.removeAllListeners();
+              window.ga4PageViewListener = undefined;
+            }
+          },
+          getCookiesToRemove: () => {
+            const getCookieOptions = () => {
+              try {
+                const domainParts = window.location.hostname.split('.')
+                const removalDomain = `.${domainParts.slice(-2).join('.')}`
+                return { domain: removalDomain, path: '/' }
+              } catch {
+                return { domain: window.location.hostname, path: '/' }
+              }
+            }
+            const options = getCookieOptions()
+            return [
+              { name: '_ga', options },
+              { name: '_ga_{{ settings.measurementId }}', options },
+            ]
+          }
+        })
+              })
+      </script>
+    {%- endunless -%}
+    {% unless has_cmp_provider %}
+      {%- if isNew -%}
+        <script async src="https://www.googletagmanager.com/gtag/js?id={{ settings.measurementId }}"></script>
+        <script>
+              window.dataLayer = window.dataLayer || [];
```

Comment:
This and the next script are not registered. Is it intentional?

Reply:
No always register to the scriptManager and scriptManager checks for the cmp installation. Apps do not need to do so.